src/protocol_handle_stream.rs

```diff
@@ -218,8 +220,8 @@ where
         self.current_task.idle();
     }
 
-    fn handle_poll(&mut self, cx: &mut Context) {
-        Pin::new(&mut self.handle).poll(cx, &mut self.handle_context);
+    fn handle_poll(&mut self, cx: &mut Context) -> Poll<Option<()>> {
```

Comment:
How about:
```rust
fn handle_poll(&mut self, cx: &mut Context) -> bool {
    if self.need_poll {
        match Pin::new(&mut self.handle).poll(cx, &mut self.handle_context) {
            Poll::Ready(None) => self.need_poll = false,
            Poll::Ready(Some(())) => {},
            Poll::Pending => return true,
        }
    }
    false
}
```

Reply:
One more function call and unnecessary &= operation, it is better to determine whether it needs to be called on the outside.